src/lib.rs

```diff
@@ -316,10 +325,21 @@ impl<'a> ModalityProbe<'a> {
         probe_id: ProbeId,
         restart_counter: RestartCounterProvider<'a>,
     ) -> Result<&'a mut ModalityProbe<'a>, StorageSetupError> {
+        // Align memory before embedding so that all padding is filled with guard bytes
+        let padding_offset = memory.as_ptr().align_offset(align_of::<Self>());
```

Comment:
Not important for this PR, but this sort of guarding seems like a pattern that could be moved in a general way into a variant of the `embed` function (currently provided by the fixed-slice-vec crate), e.g:

```
pub fn embed_with_guard_prefix<'a, T, F, E>(destination: &'a mut [u8], guard_byte: u8, f: F) -> Result<&'a mut T, EmbedValueError<E>>
where
    F: FnOnce(&'a mut [u8]) -> Result<T, E>
```

Reply:
I'll add a note as a future improvement